document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/controller/DocumentGeneratorController.java

```diff
@@ -0,0 +1,55 @@
+package uk.gov.companieshouse.document.generator.core.controller;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.document.generator.core.models.DocumentGeneratorRequest;
+import uk.gov.companieshouse.document.generator.core.models.DocumentGeneratorResponse;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import javax.validation.Valid;
+
+import static uk.gov.companieshouse.document.generator.core.DocumentGeneratorApplication.APPLICATION_NAME_SPACE;
+
+@RestController
+public class DocumentGeneratorController {
+
+    private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    @PostMapping("/document-generate")
+    @ResponseBody
+    public ResponseEntity generateDocument(@Valid @RequestBody DocumentGeneratorRequest documentGeneratorRequest,
+                                           BindingResult result) {
+
+        DocumentGeneratorResponse response = new DocumentGeneratorResponse();
+
+        if (result.hasErrors()) {
+            LOG.error("error in request body");
+            return new ResponseEntity<>(result.getAllErrors(), HttpStatus.BAD_REQUEST);
+        }
+
+        try {
+            response = null;
+        } catch (Exception e) {
+            LOG.error(e);
+        }
+
+        if(response == null) {
+            LOG.error("no data has been returned");
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+
+        if(response.getLocation() != null) {
```

Comment:
Is this tested? I feel that it is possibly missed because of the `response` always been null.